src/components/side_panel/chart/line_bar_pie_panel/config_panel.ts

```diff
@@ -30,10 +32,12 @@ export class LineBarPieConfigPanel extends Component<Props, SpreadsheetChildEnv>
 
   private dataSeriesRanges: string[] = [];
   private labelRange: string | undefined;
+  private headerPosition: number = 1;
 
   setup() {
     this.dataSeriesRanges = this.props.definition.dataSets;
     this.labelRange = this.props.definition.labelRange;
+    this.headerPosition = this.calculateHeaderPosition();
```

Comment:
@LucasLefevre I remembered I tried to calculate it dynamically, but the problem is the label range and the data set need to be valid first, then we can do the calculation. However, the dispatch results which determine the validness are stored in the state which is not updated immediately (if it works like React in which states are updated in batch, or at least when calling `calculateHeaderPosition()` they are not updated, even when keeping the first lines of code in `calculateHeaderPosition()` which are used to return early if invalid), and it will throw errors to cause the whole app crash. Using `t-if` cannot avoid it either. 
Hence I store `headerPosition` and only update it in setup and after the dataset and label range are confirmed, to be sure that it will not cause error and I think it is enough to update the header position only after confirmation. 
Do you have any idea to make it calculate dynamically without throwing errors related to invalid dataset/label range?